AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -128,29 +140,22 @@ - (void)layout
   _spinner.position = CGPointMake(bounds.size.width/2, bounds.size.height/2);
 }
 
-- (void)didLoad
+- (void)setPlaceholderImagefromAsset:(AVAsset*)asset
 {
-  [super didLoad];
-  [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(didPlayToEnd:) name:AVPlayerItemDidPlayToEndTimeNotification object:nil];
-
-  if (_shouldBePlaying) {
-    _playerNode = [[ASDisplayNode alloc] initWithLayerBlock:^CALayer *{
-      AVPlayerLayer *playerLayer = [[AVPlayerLayer alloc] init];
-      if (!_player) {
-        _player = [AVPlayer playerWithPlayerItem:[[AVPlayerItem alloc] initWithAsset:_asset]];
-        _player.muted = _muted;
-      }
-      playerLayer.player = _player;
-      playerLayer.videoGravity = [self gravity];
-      return playerLayer;
-    }];
+  ASDN::MutexLocker l(_videoLock);
+  if (!_placeholderImageNode)
+    _placeholderImageNode = [[ASImageNode alloc] init];
```

Comment:
@appleguy Subclassing is going to be a bit beyond my ASDK knowledge if I'm honest! 

I can see where you're coming from, but looking at the external interface of ASImageNode I'm not sure all of the aspects would make sense. For instance, we'd have to implement cropRect. Is display completion relevant? Could we implement tint colours anyway?

Though I can see the benefits of code reuse I'm not sure the external 'is-a ASImageNode' relationship would make complete sense to the user of the class.